### PR TITLE
deploy/write-manifest tensorflow

### PR DESCRIPTION
**Pull request by aronatkins**
## Intent

Allow manifests/deploys of TensorFlow models.

Fixes https://github.com/rstudio/connect/issues/26884

## Type of Change

- [ ] Bug Fix            
- [x] New Feature       
- [ ] Breaking Change   

## Approach
Add entrypoints for TensorFlow commands. Removed R/Python/image arguments in situations where they do not apply (e.g. static content).

## Automated Tests
Test assembly of manifest and bundle; will add integration tests separately.

## Checklist

- [x] I have updated [CHANGELOG.md](../CHANGELOG.md) to cover notable changes.
- [x] I have updated all related GitHub issues to reflect their current state.

### CODE REVIEW

File: rsconnect/main.py [hunk 1]
```diff
@@ -1325,6 +1324,98 @@ def deploy_quarto(
...
 
 
+# noinspection SpellCheckingInspection,DuplicatedCode
+@deploy.command(
+    name="tensorflow",
+    short_help="Deploy TensorFlow models to Posit Connect [v2024.05.0+].",
+    help=(
+        "Deploy a TensorFlow model to Posit Connect. Requires Posit Connect 2024.05.0 or later."
+        "\n\n"
+        "DIRECTORY is the path containing a TensorFlow model."
```
Comment (toph-allen) [inline rsconnect/main.py:1334]: Should `DIRECTORY` be in all caps here? If this is just a convention I'm unaware of, ignore this.

File: tests/test_bundle.py [hunk 2]
```diff
@@ -2079,90 +2065,6 @@ def test_create_html_manifest():
     )
     assert single_file_index_file_ans == json.loads(manifest.get_flattened_copy().json)
 
-    # check all runtime_environment vars
-    single_file_index_file_ans = {
```
Comment (toph-allen) [inline tests/test_bundle.py:2083]: What were these tests doing?

**Review by toph-allen — APPROVED** [review]

File: tests/test_bundle.py (hunk 2 above)
Comment (aronatkins) [reply tests/test_bundle.py:2083]: These tests were all trying to confirm that R/Python environment and image fields were retained for static (HTML) content. We do not execute HTML content, so these fields are unnecessary. I removed them as inputs and removed the tests as appropriate.

File: rsconnect/main.py (hunk 1 above)
Comment (aronatkins) [reply rsconnect/main.py:1334]: The all-caps is a convention we used for Quarto content commands. Its presence feels a little redundant, but folks found it helpful during early testing.

**Pull request merged**